src/wp-admin/includes/image.php

```diff
@@ -252,6 +253,14 @@ function wp_create_image_subsizes( $file, $attachment_id ) {
 		$image_meta['image_meta'] = $exif_meta;
 	}
 
+	// Calculate the primary and additional mime types to generate.
+	$upload_mime_transforms = wp_upload_image_mime_transforms( $attachment_id );
+	$mime_type              = wp_get_image_mime( $file );
+	$mime_extension_map     = array_flip( wp_get_mime_types() );
+	$mimes_to_generate      = _wp_get_primary_and_additional_mime_types( $upload_mime_transforms, $attachment_id, $mime_type );
+	$primary_mime_type      = isset( $mimes_to_generate['primary_mime_type'] ) ? $mimes_to_generate['primary_mime_type'] : array();
+	$additional_mime_types  = isset( $mimes_to_generate['additional_mime_types'] ) ? $mimes_to_generate['additional_mime_types'] : array();
```

Comment:
This code can become a bit simpler:
```suggestion
	list( $primary_mime_type, $additional_mime_types ) = _wp_get_primary_and_additional_mime_types( $upload_mime_transforms, $attachment_id, $mime_type );
```

Also, you have `array()` as fallback for the `$primary_mime_type` here but it needs to be a string.